### PR TITLE
Use named groups instead of numbered groups to refer to clozeReg

### PR DESCRIPTION
**Pull request by agentydragon**
Named groups are less fragile than referring to groups by indices. With named groups, you can freely add new captured groups and don't have to reindex a lot of existing references to group indices.

This should not break backward compatibility, since the change keeps group indices the same as before.

### CODE REVIEW

File: anki/media.py [hunk 1]
```diff
@@ -232,23 +232,23 @@ def filesInStr(
...
 
-    def _expandClozes(self, string) -> List[str]:
+    def _expandClozes(self, string: str) -> List[str]:
         ords = set(re.findall(r"{{c(\d+)::.+?}}", string))
         strings = []
-        from anki.template.template import clozeReg
+        from anki.template.template import clozeReg, matchGroupHint, matchGroupContent
 
         def qrepl(m):
             if m.group(4):
```
Comment (dae) [inline anki/media.py]: missed this?

File: anki/template/template.py [hunk 2]
```diff
@@ -5,7 +5,17 @@
...
+# named groups:
+#   - tag: The lowercase or uppercase 'c' letter opening the Cloze.
+#   - content: Clozed-out content.
+#   - hint: Cloze hint, if provided.
+clozeReg = r"(?si)\{\{(?P<tag>c)%s::(?P<content>.*?)(::(?P<hint>.*?))?\}\}"
+
+# Constants referring to group names within clozeReg.
+matchGroupTag = "tag"
+matchGroupContent = "content"
+matchGroupHint = "hint"
```
Comment (dae) [inline anki/template/template.py]: The constants are not clearly related to clozeReg, nor clearly constants. Perhaps something like CLOZEREG_TAG or CLOZE_GROUP_TAG would be clearer?

Comment (agentydragon) [reply anki/template/template.py]: Sure, done. (I prefer `UPPER_SNAKE` for constants, too, but didn't know if it was OK to switch convention.)

File: anki/media.py (hunk 1 above)
Comment (agentydragon) [reply anki/media.py]: Thanks, fixed

File: anki/template/template.py (hunk 2 above)
Comment (dae) [reply anki/template/template.py]: Existing variables shouldn't be changed, but the contrib file was recently changed to allow PEP8 in newly contributed code.

**Comment by dae** [comment]
Thanks!

**Pull request merged**